### ISSUE TITLE
Action-Conditioned 3D Human Motion Synthesis with Transformer VAE	

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#125]
![image](https://user-images.githubusercontent.com/46308407/143995426-1726d86c-28ad-4fc4-940f-883b46fe2098.png)

## TL;DR

- Action-conditioned (e.g., "throw") 3D human motion (in SMPL format) synthesis with Transformer VAE (architecture-wise similar to Dall-E)
- Weak action labels on UESTC dataset using off-the-shelf VIBE model

## Motivation

- The task has been less explored, probably due to lack of well-organized dataset

## Previous Works

- DVGAN: short text to action
- Text2Action, Language2Pose: text-to-motion
- Action2Motion: per-frame VAE on actions using GRU

## Method

![image](https://user-images.githubusercontent.com/46308407/143995466-43039e46-84b7-4df0-ac66-5cfdb11f8dd0.png)

### Motion Representation (SMPL)

- 24 rotation nodes (24 x 6 dim): R_t
- Location parameters: D_t
- Concatenate(R_t, D_t) = P_t
- With any shape parameter, we can obtain mesh vertices and body joint coordinates using SMPL (Skinned Multi-Person Linear model)

### Transformer CVAE

**Encoder** <br>
- For each action, introduce two additional parameters (mu_token, sigma_token) that are used to output VAE mean and variance parameters
- For encoder output, only take the first two tokens (mu, sigma) and discard the rest

**Decoder** <br>
- Positional encoded time information (PE(1), PE(2), ..., PE(T)) as *query*
- Latent vector (z~N(0,1)) + Learnable action bias as *key* and *value*

### Training

- Reconstruction on Rotation Parameters
- Reconstruction on Vertices
- KL-div for VAE-regularization (bind mu=0, sigma=Identity)

## Experiments

![image](https://user-images.githubusercontent.com/46308407/143996473-36842c28-b947-4c11-bcdf-66012a9c0fed.png)
![image](https://user-images.githubusercontent.com/46308407/143996492-4e9669fd-2249-408e-bc7f-9a4cc86a9a04.png)
![image](https://user-images.githubusercontent.com/46308407/143996514-f3c3e3e3-d554-4380-b21d-a1cae28d6ad7.png)

- Varying number of time frames during training helps to generalize in inference time

![image](https://user-images.githubusercontent.com/46308407/143996563-11b80c80-fbf4-4493-8aa5-2789803f477b.png)

### Why do we need action2motion?

- Motion Denoising
![image](https://user-images.githubusercontent.com/46308407/143996585-6424cf90-21a2-4bd5-8f0d-efbc7ad73737.png)
- Data Augmentation (for action recognition)
![image](https://user-images.githubusercontent.com/46308407/143996641-a7fc4309-0a97-4be8-834f-5f9d8032261f.png)

## Comments

- Rather naive approach to action2motion using recently popular transformer+VAE framework, but good baseline for under-explored task.
- Still some limitations since they can only generate motion sequence on several *predefined* action classes.
- I think it's hard to measure quantitatively the performance of each methods. I'm not sure FID is a reliable metric in this case, in particular.
